### ISSUE TITLE
@Value.Default annotation is redundant for Java 8 default interface method

### ISSUE THREAD

**Issue by krisskross** [immutables/immutables#125]
Hi

The annotation processor API can tell if a Java 8 interface method have a 'default' method implementation or not.

Hence, the @Value.Default annotation should be allowed to be optional for Java 8 interfaces. It is still very clear that the property provide a default value since the 'default' keyword is used on the method.

Cheers,
-Kristoffer

**Comment by elucash** [comment]
Hi Kristoffer
Thank you for feature request. I thought about this for a while at the time when we were implementing java8 support. The issue with that is we then won't have a way to add regular default methods to the interface/trait without turning it into default attribute accessor and having an initializer.

```java
interface Val {
  // plain attribute
  Iterable<String> values();

  @Value.Default // used to specify initializable attribute
  default int initializeOrDefault() {
    return 1;
  }
  // regular method with no backing field
  // new stream will be returned each time
  // this would not work if `default` will imply `@Value.Default`
  // or will require additional opt-out annotation: `@Value.NonDefault`
  default Stream<String> stream() {
    return StreamSupport.stream(
        Spliterators.spliteratorUnknownSize(
            values().iterator(),
            Spliterator.ORDERED), false);
  }
  @Value.Derived // this will be computed and stored
  default int size() {
     return Iterables.size(values());
  }
}
```

You could also notice how `@Value.Default` is redundant for immutable annotations where `default` value is used. The meaning of `default` there is more or less the same as `@Value.Default`

That said, using `default` in the way you're suggesting is still attractive, but, probably should be enabled as a `@Value.Style` attribute.

**Comment by krisskross** [comment]
Yes, is see your point. A style attribute is probably the better way to do it.

Another idea is to allow naming conventions, i.e. turn all 'get*' methods into attributes and let this style automatically use 'default' implementations as @Value.Default.

**Comment by elucash** [comment]
The enhancement was released in 2.0.15.
Style attribute is called `defaultAsDefault`, it is `false` if not specified, when enabled it will treat all `default` accessor methods (in covered by style value types) as if they annotated `@Value.Default`
